crates/storage/src/map.rs

```diff
@@ -425,3 +423,1304 @@ mod test {
+        let all: StdResult<Vec<_>> = TRIPLE.range(&store, None, None, Order::Ascending).collect();
+        let all = all.unwrap();
+        assert_eq!(4, all.len());
+        assert_eq!(all, vec![
+            ((b"owner".to_vec(), 9, "recipient".to_string()), 1000),
+            ((b"owner".to_vec(), 9, "recipient2".to_string()), 3000),
+            ((b"owner".to_vec(), 10, "recipient3".to_string()), 3000),
+            ((b"owner2".to_vec(), 9, "recipient".to_string()), 5000)
+        ]);
+
+        // let's iterate over a sub_prefix
+        let all: StdResult<Vec<_>> = TRIPLE
+            .prefix(b"owner")
+            .range(&store, None, None, Order::Ascending)
+            .collect();
+        let all = all.unwrap();
+        assert_eq!(3, all.len());
+        assert_eq!(all, vec![
+            ((9, "recipient".to_string()), 1000),
+            ((9, "recipient2".to_string()), 3000),
+            ((10, "recipient3".to_string()), 3000),
+        ]);
+
+        // let's iterate over a prefix
+        let all: StdResult<Vec<_>> = TRIPLE
+            .prefix(b"owner")
+            .append(9)
+            .range(&store, None, None, Order::Ascending)
+            .collect();
+        let all = all.unwrap();
+        assert_eq!(2, all.len());
+        assert_eq!(all, vec![
+            ("recipient".to_string(), 1000),
+            ("recipient2".to_string(), 3000),
+        ]);
+
+        // let's try to iterate over a prefixed restricted inclusive range
+        let all: StdResult<Vec<_>> = TRIPLE
+            .prefix(b"owner")
+            .append(9)
+            .range(
+                &store,
+                Some(Bound::inclusive("recipient")),
+                None,
+                Order::Ascending,
+            )
+            .collect();
+        let all = all.unwrap();
+        assert_eq!(2, all.len());
+        assert_eq!(all, vec![
+            ("recipient".to_string(), 1000),
+            ("recipient2".to_string(), 3000),
+        ]);
+
+        // let's try to iterate over a prefixed restricted exclusive range
+        let all: StdResult<Vec<_>> = TRIPLE
+            .prefix(b"owner")
+            .append(9)
+            .range(
+                &store,
+                Some(Bound::exclusive("recipient")),
+                None,
+                Order::Ascending,
+            )
+            .collect();
+        let all = all.unwrap();
+        assert_eq!(1, all.len());
+        assert_eq!(all, vec![("recipient2".to_string(), 3000),]);
+    }
+
+    #[test]
+    fn basic_update() {
+        let mut store = MockStorage::new();
+
+        let add_ten = |a: Option<u64>| -> StdResult<_> { Ok(Some(a.unwrap_or_default() + 10)) };
+
+        // save and load on three keys, one under different owner
+        let key: (&[u8], &[u8]) = (b"owner", b"spender");
+        ALLOWANCE.update(&mut store, key, add_ten).unwrap();
+        let twenty = ALLOWANCE.update(&mut store, key, add_ten).unwrap().unwrap();
+        assert_eq!(20, twenty);
```

Comment:
Also for consistency, in `assert_eq!` let's always put the actual value first, expected value second.